src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -1670,6 +1679,12 @@ void WorldSession::HandleCharFactionOrRaceChange(WorldPacket& recvData)
         return;
     }
 
+    //Prevent To rename Character
+    if ((factionChangeInfo.Name != oldName) && (sWorld->getIntConfig(CONFIG_PERVENT_RENAME_CUSTOMIZATION) == 1))
+    {
+        SendCharFactionChange(CHAR_NAME_DECLENSION_DOESNT_MATCH_BASE_NAME, factionChangeInfo);
```

Comment:
You should pick a better error code here - this one is specifically for russian clients